modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PrepareServiceImpl.java

```diff
@@ -142,24 +156,39 @@ public void stop() throws Exception {
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<QueryPlan> prepareAsync(ParsedResult parsedResult, BaseQueryContext ctx) {
-        try {
-            PlanningContext planningContext = PlanningContext.builder()
-                    .parentContext(ctx)
-                    .build();
+        CompletableFuture<QueryPlan> result;
+
+        PlanningContext planningContext = PlanningContext.builder()
+                .parentContext(ctx)
+                .query(ctx.query())
+                .plannerTimeout(plannerTimeout)
+                .build();
+
+        result = prepareAsync0(parsedResult, planningContext);
+
+        return result.exceptionally(ex -> {
+                    Throwable th = ExceptionUtils.unwrapCause(ex);
+                    if (planningContext.timeouted() && th instanceof RelOptPlanner.CannotPlanException) {
+                        LOG.info("Query plan is absent due to planing timeout is reached [query={}]", ctx.query());
```

Comment:
this is the second line in the log about planning timeout.

Reply:
removed